File: competitions/CookOff/tandom.py
inp = raw_input()
l = len(inp)

def isTandom(inp):
	l1 = len(inp)
	size = l1 / 3

	if(inp[:size] == inp[size: 2 * size] == inp[2 * size:]):
		return True
	else:
		return False

countint = 0
countbore = 0

for i in range(l - 2):
	for j in range(i + 2, l, 3):
		if isTandom(inp[i:j + 1]):
			try:
				if inp[i] != inp[j + 2]:
					countint = countint + 1
				else:
					countbore = countbore + 1
			except:
				countint = countint + 1

print countint, countbore

File: competitions/CookOff/tandom.cpp
#include<bits/stdc++.h>

using namespace std;

bool isTandom(string inp)
{
	int l = inp.length();
	int size = l / 3;

	if(inp.substr(0, size).compare(inp.substr(size, size)) == 0)
		if(inp.substr(size, size).compare(inp.substr(2 * size, size) ) == 0)
			return true;

	return false;	

}

int main()
{
	string inp;
	cin>>inp;

	int countint, countbore;
	countint = countbore = 0;

	for(int i = 0; i < inp.length() - 2; i++)
	{
		for(int j = 1; j * 3 <= inp.length() - i; j ++)
		{			
			if (isTandom(inp.substr(i, j * 3)))
			{
				if (j + 2 > inp.length())
					countint++;
				else if (inp[i] != inp[i + j * 3])
					countint++;
				else
					countbore++;
			}
		}
	}

	cout<<countint<<" "<<countbore<<endl;
	return 0;	
}